### PR TITLE
feat: deploy solana, eclipsemainnet

### PR DESCRIPTION
**Pull request by daniel-savu**

### CODE REVIEW

File: rust/config/mainnet_config.json [hunk 1]
```diff
@@ -1866,6 +1866,91 @@
...
+      },
+      "chainId": 9571022593,
+      "deployer": {
+        "name": "Abacus Works",
+        "url": "https://www.hyperlane.xyz"
+      },
+      "displayName": "Eclipse",
+      "domainId": 9571022593,
+      "index": {
+        "from": 727595,
```
Comment (tkporter) [inline rust/config/mainnet_config.json:1934]: this is 1 on mainnet2, I kinda would've thought it'd be 0, but probably worth setting that https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/v2/rust/config/mainnet_config.json#L161

File: typescript/infra/config/environments/mainnet3/aw-validators/hyperlane.json [hunk 2]
```diff
@@ -44,10 +50,21 @@
...
   "cheesechain": {
-    "validators": ["0x478fb53c6860ae8fc35235ba0d38d49b13128226"]
+    "validators": [
+      "0x478fb53c6860ae8fc35235ba0d38d49b13128226"
+    ]
+  },
+  "eclipse": {
+    "validators": [
+      "0x54c96aa1666596c043b07d694da552aa4b596306",
+      "0xb6c6f6e2aa92dc586ea7a7b4c303ef9a6f4dd4a4",
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/aw-validators/hyperlane.json:60]: we should just run 1

File: typescript/infra/config/environments/mainnet3/validators.ts [hunk 3]
```diff
@@ -400,6 +400,38 @@ export const validatorChainConfig = (
...
+      ),
+    },
+    eclipse: {
+      interval: 5,
+      reorgPeriod: getReorgPeriod('eclipse'),
+      validators: validatorsConfig(
+        {
+          [Contexts.Hyperlane]: [
+            '0x54c96aa1666596c043b07d694da552aa4b596306',
+            '0xb6c6f6e2aa92dc586ea7a7b4c303ef9a6f4dd4a4',
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/validators.ts:426]: rm

File: typescript/sdk/src/consts/multisigIsm.ts [hunk 4]
```diff
@@ -351,6 +351,23 @@ export const defaultMultisigConfigs: ChainMap<MultisigConfig> = {
     ],
   },
 
+  eclipse: {
+    threshold: 2,
+    validators: [
+      '0x54c96aa1666596c043b07d694da552aa4b596306',
+      '0xb6c6f6e2aa92dc586ea7a7b4c303ef9a6f4dd4a4',
```
Comment (tkporter) [inline typescript/sdk/src/consts/multisigIsm.ts:358]: rm

File: typescript/infra/chain-config.json [hunk 5]
```diff
@@ -0,0 +1,1429 @@
+{
```
Comment (tkporter) [inline typescript/infra/chain-config.json:1]: did you mean to check this in? seems like a mostly dupe of `rust/sealevel/environments/mainnet3/chain-config.json`

File: typescript/infra/config/environments/mainnet3/agent.ts [hunk 6]
```diff
@@ -138,6 +142,7 @@ export const hyperlaneContextAgentChainConfig: AgentChainConfig<
     bsc: true,
     celo: true,
     cheesechain: true,
+    eclipse: false,
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/agent.ts:145]: not critical but there has been confusion in the past why these are set to false, maybe the `// Cannot scrape non-EVM chains` comment is worth it for this and solana

File: typescript/infra/config/environments/mainnet3/aw-validators/hyperlane.json [hunk 7]
```diff
@@ -1,6 +1,8 @@
 {
   "ancient8": {
-    "validators": ["0xbb5842ae0e05215b53df4787a29144efb7e67551"]
+    "validators": [
+      "0xbb5842ae0e05215b53df4787a29144efb7e67551"
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/aw-validators/hyperlane.json:4]: probably need a prettier run

File: typescript/infra/config/environments/mainnet3/funding.ts [hunk 8]
```diff
@@ -58,10 +58,12 @@ export const keyFunderConfig: KeyFunderConfig<
     xlayer: '0.5',
     zetachain: '20',
     zoramainnet: '0.2',
-    // ignore cosmos chains
+    // ignore non-evm chains
```
Comment (tkporter) [inline typescript/infra/config/environments/mainnet3/funding.ts:61]: inclusive 🤝

**Review by tkporter — APPROVED** [review]

**Review by paulbalaji — APPROVED** [review]

**Pull request merged**